src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -382,4 +385,55 @@ private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
 
+  @Test
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000,
+        "foobared", 0, "my_shiny_client_name") { };
+
+    try (JedisPool pool = new JedisPool(new JedisPoolConfig(), factory)) {
+      Jedis obj1;
+      try (Jedis obj11 = pool.getResource()) {
+        obj1 = obj11;
+        obj11.set("foo", "bar");
+        assertEquals("bar", obj11.get("foo"));
+        assertEquals(1, pool.getNumActive());
+      }
+      assertEquals(0, pool.getNumActive());
+      Jedis obj12 = pool.getResource();
```

Comment:
`       Jedis obj12 = pool.getResource(); ` can be put in a `try` resource block